platforms/nuttx/cmake/init.cmake

```diff
@@ -73,57 +73,61 @@ endif()
 
-# copy during cmake configure
-execute_process(COMMAND ${cp_cmd} ${cp_opts} ${CP_SRC} ${CP_DST} WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
-
-###############################################################################
-# nuttx-config: copy to build directory
-###############################################################################
-
-set(NUTTX_DEFCONFIG ${NUTTX_CONFIG_DIR}/${NUTTX_CONFIG}/defconfig CACHE FILEPATH "path to defconfig" FORCE)
+# copy apps during cmake configure
+if(NOT EXISTS ${NUTTX_DIR}/apps_copy.stamp)
+	execute_process(COMMAND ${NUTTX_COPY_CMD} ${NUTTX_COPY_CMD_OPTS} ${CP_SRC} ${CP_DST} WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
+	execute_process(COMMAND ${CMAKE_COMMAND} -E touch ${PX4_BINARY_DIR}/NuttX/apps_copy.stamp)
+endif()
 
 ###############################################################################
-# Create a temporary Toplevel Make.defs for the oldconfig step
+# board nuttx-config
 ###############################################################################
-configure_file(${NUTTX_SRC_DIR}/Make.defs.in ${NUTTX_DIR}/Make.defs)
 
 # If the board provides a Kconfig Use it or create an empty one
-if(EXISTS ${NUTTX_CONFIG_DIR}/Kconfig)
-	execute_process(COMMAND ${CMAKE_COMMAND} -E copy_if_different ${NUTTX_CONFIG_DIR}/Kconfig ${NUTTX_DIR}/boards/dummy/Kconfig)
-else()
-	execute_process(COMMAND ${CMAKE_COMMAND} -E touch ${NUTTX_DIR}/boards/dummy/Kconfig)
+if(NOT EXISTS ${NUTTX_DIR}/nuttx_config_kconfig.stamp)
+	if(EXISTS ${NUTTX_CONFIG_DIR}/Kconfig)
+		execute_process(COMMAND ${CMAKE_COMMAND} -E copy_if_different ${NUTTX_CONFIG_DIR}/Kconfig ${NUTTX_DIR}/boards/dummy/Kconfig)
+	else()
+		execute_process(COMMAND ${CMAKE_COMMAND} -E touch ${NUTTX_DIR}/boards/dummy/Kconfig)
+	endif()
+
+	execute_process(COMMAND ${CMAKE_COMMAND} -E touch ${PX4_BINARY_DIR}/NuttX/nuttx_config_kconfig.stamp)
 endif()
 
-execute_process(
-	COMMAND ${CMAKE_COMMAND} -E make_directory ${PX4_BINARY_DIR}/NuttX/nuttx-config/src
-	COMMAND ${CMAKE_COMMAND} -E copy_directory ${NUTTX_CONFIG_DIR}/ ${PX4_BINARY_DIR}/NuttX/nuttx-config
-)
+if(NOT EXISTS ${NUTTX_DIR}/nuttx_copy_config_dir.stamp)
 
-# NuttX extra files
-execute_process(COMMAND ${CMAKE_COMMAND} -E copy_if_different ${NUTTX_SRC_DIR}/nsh_romfsimg.h ${PX4_BINARY_DIR}/NuttX/nuttx-config/include/nsh_romfsimg.h)
+	# copy board's nuttx-config to NuttX/nuttx-config
+	file(RELATIVE_PATH CP_SRC ${CMAKE_SOURCE_DIR} ${PX4_BOARD_DIR}/nuttx-config)
+	file(RELATIVE_PATH CP_DST ${CMAKE_SOURCE_DIR} ${PX4_BINARY_DIR}/NuttX)
+	execute_process(COMMAND ${NUTTX_COPY_CMD} ${NUTTX_COPY_CMD_OPTS} ${CP_SRC} ${CP_DST} WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
 
-# copy defconfig
-execute_process(COMMAND ${CMAKE_COMMAND} -E copy_if_different ${NUTTX_DEFCONFIG} ${NUTTX_DIR}/.config)
+	execute_process(COMMAND ${CMAKE_COMMAND} -E make_directory ${PX4_BINARY_DIR}/NuttX/nuttx-config/drivers)
+	execute_process(COMMAND ${CMAKE_COMMAND} -E touch ${PX4_BINARY_DIR}/NuttX/nuttx-config/drivers/Kconfig)
```

Comment:
Need to deal with existing drivers/Kconfig)

Reply:
Also - not sure If I am ready to loos the uncompressed config option - is it a problem or just messy?

Reply:
It was an untested path I didn't think about handling. It could probably be updated and kept if needed. Isn't this in line with current NuttX where only the compressed version is stored back in the board config? Where would we still use this? We've already had the ability to migrate out in a PX4 release for a while, so I think we're safe to move on.

Reply:
Fare enough!